client/src/main/java/com/spectralogic/tpfr/client/model/TimeCode.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * ***************************************************************************
+ *   Copyright 2016-2017 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ***************************************************************************
+ */
+
+package com.spectralogic.tpfr.client.model
+
+import java.util.regex.Pattern
+
+class TimeCode(val time: String) {
+
+    init {
+        if (!isValidFormat(time)) {
```

Comment:
Rather than doing validation in a constructor, create a Companion Object factory method (i.e. static method) that will perform the validation, and then create the TimeCode instance.  Also make the constructor private in this case.